FreeOrion (0.5 and Git master) fails to build from source on 32 bit architectures e.g. i386

Bug Report
==========

Hi,

Debian maintainer of FreeOrion here. Version 0.5 and latest Git master fail to build from source on i386 and other 32 bit architectures.

https://buildd.debian.org/status/package.php?p=freeorion

I thought this issue was related to https://github.com/freeorion/freeorion/issues/4583

The problem appears to be in Meter.cpp.

https://buildd.debian.org/status/fetch.php?pkg=freeorion&arch=i386&ver=0.5%2Bgit20230820-1&stamp=1692559465&raw=0

````
[ 15%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Meter.cpp.o
/usr/bin/c++ -DBINPATH=\"/usr/bin\" -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DENABLE_BINRELOC -DFREEORION_BUILD_COMMON -DFREEORION_LINUX -DSHAREPATH=\"/usr/share\" -Dfreeorioncommon_EXPORTS -isystem /usr/include/python3.11 -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -O2 -g -DNDEBUG -std=c++20 -fPIC -fvisibility=hidden -Wall -MD -MT CMakeFiles/freeorioncommon.dir/universe/Meter.cpp.o -MF CMakeFiles/freeorioncommon.dir/universe/Meter.cpp.o.d -o CMakeFiles/freeorioncommon.dir/universe/Meter.cpp.o -c /<<PKGBUILDDIR>>/universe/Meter.cpp
/<<PKGBUILDDIR>>/universe/Meter.cpp:20:32: error: static assertion failed
   20 |     static_assert(r2.Current() == 64999.001f);
      |                   ~~~~~~~~~~~~~^~~~~~~~~~~~~
/<<PKGBUILDDIR>>/universe/Meter.cpp:20:32: note: the comparison reduces to ‘(6.4999e+4l == 6.49990010000000000012e+4l)’
/<<PKGBUILDDIR>>/universe/Meter.cpp:26:32: error: static assertion failed
   26 |     static_assert(r3.Current() == 64998.991f);
      |                   ~~~~~~~~~~~~~^~~~~~~~~~~~~
/<<PKGBUILDDIR>>/universe/Meter.cpp:26:32: note: the comparison reduces to ‘(6.499898828125e+4l == 6.49989909999999999997e+4l)’
/<<PKGBUILDDIR>>/universe/Meter.cpp:40:32: error: static assertion failed
   40 |     static_assert(q2.Current() == 1.01f);
      |                   ~~~~~~~~~~~~~^~~~~~~~
/<<PKGBUILDDIR>>/universe/Meter.cpp:40:32: note: the comparison reduces to ‘(1.00999999046325683594e+0l == 1.00999999999999999999e+0l)’
/<<PKGBUILDDIR>>/universe/Meter.cpp:46:32: error: static assertion failed
   46 |     static_assert(q3.Current() == 0.01f);
      |                   ~~~~~~~~~~~~~^~~~~~~~
/<<PKGBUILDDIR>>/universe/Meter.cpp:46:32: note: the comparison reduces to ‘(9.99999977648258209229e-3l == 9.9999999999999999998e-3l)’
make[3]: *** [CMakeFiles/freeorioncommon.dir/build.make:499: CMakeFiles/freeorioncommon.dir/universe/Meter.cpp.o] Error 1
make[3]: Leaving directory '/<<PKGBUILDDIR>>/obj-i686-linux-gnu'
make[2]: *** [CMakeFiles/Makefile2:496: CMakeFiles/freeorioncommon.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
````

Environment
-----------

Debian Sid
FreeOrion 0.5 and Git master

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,7 +60,6 @@ include(CPackComponent)
 
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 
 option(BUILD_SHARED_LIBS "Build project as shared libraries." ON)

--- a/GG/CMakeLists.txt
+++ b/GG/CMakeLists.txt
@@ -30,7 +30,6 @@ include(Util)
 set(GIGI_VERSION 0.8.0)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
 
 option(ENABLE_PNG_TEXTURES "Enable PNG texture support." ON)
 option(ENABLE_TIFF_TEXTURES "Enable TIFF texture support." OFF)

